From bb1d8f73182e2d479309cef575e69a4ac735045c Mon Sep 17 00:00:00 2001
From: Michael Biebl <biebl@debian.org>
Date: Tue, 13 Apr 2010 04:59:21 +0200
Subject: [PATCH] Improve UNAC linking

Improve PANGO linking

Improve DEVICEKIT_POWER linking

Improve HAL linking

Improve GDKPIXBUF linking
---
 docs/reference/libtracker-client/Makefile.am |    8 +-------
 docs/reference/libtracker-common/Makefile.am |    8 +-------
 docs/reference/libtracker-miner/Makefile.am  |    8 +-------
 examples/libtracker-miner/Makefile.am        |    8 +-------
 src/libtracker-common/Makefile.am            |   10 ++--------
 src/libtracker-extract/Makefile.am           |    2 --
 src/libtracker-fts/Makefile.am               |    2 ++
 src/libtracker-miner/Makefile.am             |   15 ++-------------
 src/miners/fs/Makefile.am                    |    2 --
 src/tracker-search-bar/Makefile.am           |    2 ++
 src/tracker-search-tool/Makefile.am          |    3 +++
 src/tracker-status-icon/Makefile.am          |    2 ++
 src/tracker-store/Makefile.am                |    2 --
 tests/libtracker-common/Makefile.am          |    1 -
 tests/libtracker-extract/Makefile.am         |    1 -
 tests/libtracker-miner/Makefile.am           |    1 -
 tests/tracker-extract/Makefile.am            |    1 -
 tests/tracker-miner-fs/Makefile.am           |    2 --
 tests/tracker-store/Makefile.am              |    1 -
 19 files changed, 17 insertions(+), 62 deletions(-)

diff --git a/docs/reference/libtracker-client/Makefile.am b/docs/reference/libtracker-client/Makefile.am
index 9c2c945..522a69c 100644
--- a/docs/reference/libtracker-client/Makefile.am
+++ b/docs/reference/libtracker-client/Makefile.am
@@ -33,18 +33,12 @@ INCLUDES = 								\
 	-I$(top_srcdir)/src						\
 	$(GLIB2_CFLAGS)							\
 	$(GIO_CFLAGS)							\
-	$(PANGO_CFLAGS)							\
-	$(HAL_CFLAGS)							\
-	$(DBUS_CFLAGS)							\
-	$(UNAC_CFLAGS)
+	$(DBUS_CFLAGS)
 
 GTKDOC_LIBS =								\
 	$(top_builddir)/src/libtracker-client/libtracker-client-@TRACKER_API_VERSION@.la \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
-	$(HAL_LIBS)							\
 	$(DBUS_LIBS)							\
-	$(UNAC_LIBS)							\
-	$(PANGO_LIBS)							\
 	$(GIO_LIBS)							\
 	$(GLIB2_LIBS)
 
diff --git a/docs/reference/libtracker-common/Makefile.am b/docs/reference/libtracker-common/Makefile.am
index e8cc890..1c77b36 100644
--- a/docs/reference/libtracker-common/Makefile.am
+++ b/docs/reference/libtracker-common/Makefile.am
@@ -31,17 +31,11 @@ INCLUDES = 								\
 	-I$(top_srcdir)/src						\
 	$(GLIB2_CFLAGS)							\
 	$(GIO_CFLAGS)							\
-	$(PANGO_CFLAGS)							\
-	$(HAL_CFLAGS)							\
-	$(DBUS_CFLAGS)							\
-	$(UNAC_CFLAGS)
+	$(DBUS_CFLAGS)
 
 GTKDOC_LIBS =								\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
-	$(HAL_LIBS)							\
 	$(DBUS_LIBS)							\
-	$(UNAC_LIBS)							\
-	$(PANGO_LIBS)							\
 	$(GIO_LIBS)							\
 	$(GLIB2_LIBS)
 
diff --git a/docs/reference/libtracker-miner/Makefile.am b/docs/reference/libtracker-miner/Makefile.am
index 7e4dbe6..650b6f3 100644
--- a/docs/reference/libtracker-miner/Makefile.am
+++ b/docs/reference/libtracker-miner/Makefile.am
@@ -42,19 +42,13 @@ INCLUDES = 								\
 	-I$(top_srcdir)/src						\
 	$(GLIB2_CFLAGS)							\
 	$(GIO_CFLAGS)							\
-	$(PANGO_CFLAGS)							\
-	$(HAL_CFLAGS)							\
-	$(DBUS_CFLAGS)							\
-	$(UNAC_CFLAGS)
+	$(DBUS_CFLAGS)
 
 GTKDOC_LIBS =								\
 	$(top_builddir)/src/libtracker-miner/libtracker-miner-@TRACKER_API_VERSION@.la	 \
 	$(top_builddir)/src/libtracker-client/libtracker-client-@TRACKER_API_VERSION@.la \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
-	$(HAL_LIBS)							\
 	$(DBUS_LIBS)							\
-	$(UNAC_LIBS)							\
-	$(PANGO_LIBS)							\
 	$(GIO_LIBS)							\
 	$(GLIB2_LIBS)
 
diff --git a/examples/libtracker-miner/Makefile.am b/examples/libtracker-miner/Makefile.am
index 5f753ce..72300ed 100644
--- a/examples/libtracker-miner/Makefile.am
+++ b/examples/libtracker-miner/Makefile.am
@@ -4,18 +4,12 @@ INCLUDES =							\
 	-DSHAREDIR=\""$(datadir)"\"				\
 	-DG_LOG_DOMAIN=\"Tracker\"				\
 	-DTRACKER_COMPILATION					\
-	-DI_KNOW_THE_DEVICEKIT_POWER_API_IS_SUBJECT_TO_CHANGE	\
 	-I$(top_srcdir)/src					\
 	$(WARN_CFLAGS)						\
 	$(GLIB2_CFLAGS)						\
 	$(GCOV_CFLAGS)						\
-	$(GDKPIXBUF_CFLAGS)					\
 	$(GIO_CFLAGS)						\
-	$(PANGO_CFLAGS)						\
-	$(HAL_CFLAGS)						\
-	$(DEVKIT_POWER_CFLAGS)					\
-	$(DBUS_CFLAGS)						\
-	$(UNAC_CFLAGS)
+	$(DBUS_CFLAGS)
 
 noinst_PROGRAMS = tracker-miner-test
 
diff --git a/src/libtracker-common/Makefile.am b/src/libtracker-common/Makefile.am
index a10a2d4..5a3d33e 100644
--- a/src/libtracker-common/Makefile.am
+++ b/src/libtracker-common/Makefile.am
@@ -9,13 +9,10 @@ INCLUDES =						\
 	$(WARN_CFLAGS)					\
 	$(GLIB2_CFLAGS)					\
 	$(GCOV_CFLAGS)					\
-	$(GDKPIXBUF_CFLAGS)				\
 	$(GIO_CFLAGS)					\
-	$(PANGO_CFLAGS)					\
 	$(HAL_CFLAGS)					\
 	$(DEVKIT_POWER_CFLAGS)				\
-	$(DBUS_CFLAGS)					\
-	$(UNAC_CFLAGS)
+	$(DBUS_CFLAGS)
 
 libtracker_commondir = $(libdir)/tracker-$(TRACKER_API_VERSION)
 libtracker_common_LTLIBRARIES = libtracker-common.la
@@ -82,12 +79,9 @@ libtracker_common_la_LIBADD = 				\
 	$(HAL_LIBS)					\
 	$(DEVKIT_POWER_LIBS)				\
 	$(DBUS_LIBS)					\
-	$(UNAC_LIBS)					\
-	$(PANGO_LIBS)					\
 	$(GIO_LIBS)					\
 	$(GCOV_LIBS)					\
-	$(GLIB2_LIBS)					\
-	$(GDKPIXBUF_LIBS)
+	$(GLIB2_LIBS)
 
 marshal_sources =                              		\
         tracker-marshal.h              			\
diff --git a/src/libtracker-extract/Makefile.am b/src/libtracker-extract/Makefile.am
index 1cf67f7..8d02c20 100644
--- a/src/libtracker-extract/Makefile.am
+++ b/src/libtracker-extract/Makefile.am
@@ -8,7 +8,6 @@ INCLUDES =						\
 	$(WARN_CFLAGS)					\
 	$(GLIB2_CFLAGS)					\
 	$(GCOV_CFLAGS)					\
-	$(GDKPIXBUF_CFLAGS)				\
 	$(GIO_CFLAGS)					\
 	$(LIBIPTCDATA_CFLAGS)				\
 	$(LIBEXIF_CFLAGS) 				\
@@ -48,7 +47,6 @@ libtracker_extract_@TRACKER_API_VERSION@_la_LIBADD = 	\
 	$(GIO_LIBS)					\
 	$(GCOV_LIBS)					\
 	$(GLIB2_LIBS)					\
-	$(GDKPIXBUF_LIBS)				\
 	$(LIBIPTCDATA_LIBS)				\
 	$(LIBEXIF_LIBS)					\
 	$(EXEMPI_LIBS)
diff --git a/src/libtracker-fts/Makefile.am b/src/libtracker-fts/Makefile.am
index efe5757..4938097 100644
--- a/src/libtracker-fts/Makefile.am
+++ b/src/libtracker-fts/Makefile.am
@@ -9,6 +9,7 @@ INCLUDES =								\
 	$(GCOV_CFLAGS)							\
 	$(PANGO_CFLAGS)							\
 	$(DBUS_CFLAGS)							\
+	$(UNAC_CFLAGS)							\
 	$(SQLITE3_CFLAGS)
 
 noinst_LTLIBRARIES = libtracker-fts.la
@@ -30,4 +31,5 @@ libtracker_fts_la_LIBADD =						\
 	$(GTHREAD_LIBS)							\
 	$(GCOV_LIBS)							\
 	$(PANGO_LIBS)							\
+	$(UNAC_LIBS)							\
 	$(GLIB2_LIBS)							
diff --git a/src/libtracker-miner/Makefile.am b/src/libtracker-miner/Makefile.am
index 64cef25..a438ad6 100644
--- a/src/libtracker-miner/Makefile.am
+++ b/src/libtracker-miner/Makefile.am
@@ -5,19 +5,13 @@ INCLUDES =						\
 	-DDBUS_SERVICES_DIR=\""$(DBUS_SERVICES_DIR)"\"	\
 	-DG_LOG_DOMAIN=\"Tracker\"			\
 	-DTRACKER_COMPILATION				\
-	-DI_KNOW_THE_DEVICEKIT_POWER_API_IS_SUBJECT_TO_CHANGE	\
 	-DTRACKER_MINERS_DIR=\""$(datadir)/tracker/miners"\" 	\
 	-I$(top_srcdir)/src				\
 	$(WARN_CFLAGS)					\
 	$(GLIB2_CFLAGS)					\
 	$(GCOV_CFLAGS)					\
-	$(GDKPIXBUF_CFLAGS)				\
 	$(GIO_CFLAGS)					\
-	$(PANGO_CFLAGS)					\
-	$(HAL_CFLAGS)					\
-	$(DEVKIT_POWER_CFLAGS)				\
-	$(DBUS_CFLAGS)					\
-	$(UNAC_CFLAGS)
+	$(DBUS_CFLAGS)
 
 lib_LTLIBRARIES = libtracker-miner-@TRACKER_API_VERSION@.la
 
@@ -62,15 +56,10 @@ libtracker_miner_@TRACKER_API_VERSION@_la_LDFLAGS = 	\
 libtracker_miner_@TRACKER_API_VERSION@_la_LIBADD = 	\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la \
 	$(top_builddir)/src/libtracker-client/libtracker-client-@TRACKER_API_VERSION@.la \
-	$(HAL_LIBS)					\
-	$(DEVKIT_POWER_LIBS)				\
 	$(DBUS_LIBS)					\
-	$(UNAC_LIBS)					\
-	$(PANGO_LIBS)					\
 	$(GIO_LIBS)					\
 	$(GCOV_LIBS)					\
-	$(GLIB2_LIBS)					\
-	$(GDKPIXBUF_LIBS)
+	$(GLIB2_LIBS)
 
 if HAVE_LIBINOTIFY
 libtracker_miner_@TRACKER_API_VERSION@_la_LIBADD +=	\
diff --git a/src/miners/fs/Makefile.am b/src/miners/fs/Makefile.am
index eaa3ee5..b66035d 100644
--- a/src/miners/fs/Makefile.am
+++ b/src/miners/fs/Makefile.am
@@ -11,7 +11,6 @@ INCLUDES =								\
 	-I$(top_srcdir)/src						\
 	$(WARN_CFLAGS)							\
 	$(GMODULE_CFLAGS)						\
-	$(PANGO_CFLAGS)							\
 	$(DBUS_CFLAGS)							\
 	$(GCOV_CFLAGS)
 
@@ -37,7 +36,6 @@ tracker_miner_fs_LDADD =						\
 	$(DBUS_LIBS)							\
 	$(GMODULE_LIBS)							\
 	$(GTHREAD_LIBS)							\
-	$(PANGO_LIBS)							\
 	$(GIO_LIBS)							\
 	$(GCOV_LIBS)							\
 	$(GLIB2_LIBS)							\
diff --git a/src/tracker-search-bar/Makefile.am b/src/tracker-search-bar/Makefile.am
index fb68259..7a7fe3d 100644
--- a/src/tracker-search-bar/Makefile.am
+++ b/src/tracker-search-bar/Makefile.am
@@ -18,6 +18,7 @@ tracker_search_bar_CFLAGS = \
 	-I$(top_srcdir)/src					\
 	$(TRACKER_APPS_CFLAGS)					\
 	$(TRACKER_APPLETS_CFLAGS)				\
+	$(GDKPIXBUF_CFLAGS)					\
 	$(WARN_CFLAGS)						\
 	$(GCOV_CFLAGS)
 
@@ -25,6 +26,7 @@ tracker_search_bar_LDADD = 					\
 	$(top_builddir)/src/libtracker-client/libtracker-client-@TRACKER_API_VERSION@.la \
 	$(TRACKER_APPS_LIBS)					\
 	$(TRACKER_APPLETS_LIBS)					\
+	$(GDKPIXBUF_LIBS)					\
 	$(GCOV_LIBS)
 
 uidir = $(datadir)/tracker
diff --git a/src/tracker-search-tool/Makefile.am b/src/tracker-search-tool/Makefile.am
index 35a9225..9e9341a 100644
--- a/src/tracker-search-tool/Makefile.am
+++ b/src/tracker-search-tool/Makefile.am
@@ -25,6 +25,8 @@ tracker_search_tool_CFLAGS = \
 	-DSRCDIR=\"$(abs_srcdir)/\"				\
 	$(TRACKER_APPS_CFLAGS)					\
 	$(TRACKER_VALA_CFLAGS)					\
+	$(PANGO_CFLAGS)						\
+	$(GDKPIXBUF_CFLAGS)					\
 	$(WARN_CFLAGS)						\
 	$(GCOV_CFLAGS)						\
 	-include$(top_srcdir)/config.h  			\
@@ -33,6 +35,7 @@ tracker_search_tool_CFLAGS = \
 tracker_search_tool_LDADD = 					\
 	$(TRACKER_APPS_LIBS)					\
 	$(TRACKER_VALA_LIBS)					\
+	$(GDKPIXBUF_LIBS)					\
 	$(GCOV_LIBS)						\
 	$(NULL)
 
diff --git a/src/tracker-status-icon/Makefile.am b/src/tracker-status-icon/Makefile.am
index 82bc39c..f60e21f 100644
--- a/src/tracker-status-icon/Makefile.am
+++ b/src/tracker-status-icon/Makefile.am
@@ -8,6 +8,7 @@ INCLUDES =								\
 	-DTRACKER_COMPILATION						\
 	-I$(top_srcdir)/src						\
 	$(TRACKER_APPS_CFLAGS)						\
+	$(PANGO_CFLAGS)							\
 	$(WARN_CFLAGS)							\
 	$(GCOV_CFLAGS)
 
@@ -19,6 +20,7 @@ tracker_status_icon_LDADD = 						\
 	$(top_builddir)/src/libtracker-client/libtracker-client-@TRACKER_API_VERSION@.la \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la 	\
 	$(TRACKER_APPS_LIBS)						\
+	$(PANGO_LIBS)							\
 	$(GCOV_LIBS)							\
 	-lX11
 
diff --git a/src/tracker-store/Makefile.am b/src/tracker-store/Makefile.am
index 3481179..9aa80d2 100644
--- a/src/tracker-store/Makefile.am
+++ b/src/tracker-store/Makefile.am
@@ -15,7 +15,6 @@ INCLUDES =								\
 	$(GIO_CFLAGS)							\
 	$(GCOV_CFLAGS)							\
 	$(DBUS_CFLAGS)							\
-	$(PANGO_CFLAGS)							\
 	$(FAM_CFLAGS)							\
 	$(SQLITE3_CFLAGS)
 
@@ -65,7 +64,6 @@ tracker_store_LDADD =							\
 	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(SQLITE3_LIBS)							\
 	$(DBUS_LIBS)							\
-	$(PANGO_LIBS)							\
 	$(GIO_LIBS)							\
 	$(GMODULE_LIBS)							\
 	$(GTHREAD_LIBS)							\
diff --git a/tests/libtracker-common/Makefile.am b/tests/libtracker-common/Makefile.am
index ca1203b..76fc1fe 100644
--- a/tests/libtracker-common/Makefile.am
+++ b/tests/libtracker-common/Makefile.am
@@ -18,7 +18,6 @@ INCLUDES = 								\
 	$(GCOV_CFLAGS)							\
 	$(GMODULE_CFLAGS)						\
 	$(GTHREAD_CFLAGS)						\
-	$(PANGO_CFLAGS)							\
 	$(DBUS_CFLAGS)
 
 tracker_dbus_SOURCES = 							\
diff --git a/tests/libtracker-extract/Makefile.am b/tests/libtracker-extract/Makefile.am
index ef8e4df..dbe2193 100644
--- a/tests/libtracker-extract/Makefile.am
+++ b/tests/libtracker-extract/Makefile.am
@@ -16,7 +16,6 @@ INCLUDES = 								\
 	$(GCOV_CFLAGS)							\
 	$(GMODULE_CFLAGS)						\
 	$(GTHREAD_CFLAGS)						\
-	$(PANGO_CFLAGS)							\
 	$(DBUS_CFLAGS)
 
 tracker_utils_SOURCES = \
diff --git a/tests/libtracker-miner/Makefile.am b/tests/libtracker-miner/Makefile.am
index 5df44fa..e5b251e 100644
--- a/tests/libtracker-miner/Makefile.am
+++ b/tests/libtracker-miner/Makefile.am
@@ -28,7 +28,6 @@ INCLUDES =									\
 	$(GIO_CFLAGS)								\
 	$(GMODULE_CFLAGS)							\
 	$(GTHREAD_CFLAGS)							\
-	$(PANGO_CFLAGS)								\
 	$(DBUS_CFLAGS)
 
 LIBS =										\
diff --git a/tests/tracker-extract/Makefile.am b/tests/tracker-extract/Makefile.am
index e686e9e..0eb3a28 100644
--- a/tests/tracker-extract/Makefile.am
+++ b/tests/tracker-extract/Makefile.am
@@ -35,7 +35,6 @@ INCLUDES = 								\
 	$(GCOV_CFLAGS)							\
 	$(GMODULE_CFLAGS)						\
 	$(GTHREAD_CFLAGS)						\
-	$(PANGO_CFLAGS)							\
 	$(DBUS_CFLAGS)
 
 # TEST_UTILS_SOURCES = 							\
diff --git a/tests/tracker-miner-fs/Makefile.am b/tests/tracker-miner-fs/Makefile.am
index c3e559d..e407126 100644
--- a/tests/tracker-miner-fs/Makefile.am
+++ b/tests/tracker-miner-fs/Makefile.am
@@ -20,7 +20,6 @@ INCLUDES = 									\
 	$(GCOV_CFLAGS)								\
 	$(GMODULE_CFLAGS)							\
 	$(GTHREAD_CFLAGS)							\
-	$(PANGO_CFLAGS)								\
 	$(DBUS_CFLAGS)
 
 tracker_metadata_utils_SOURCES = 						\
@@ -35,7 +34,6 @@ tracker_metadata_utils_LDADD =	                                        	\
 	$(top_builddir)/src/libtracker-client/libtracker-client-@TRACKER_API_VERSION@.la \
 	$(top_builddir)/tests/common/libtracker-testcommon.la           	\
 	$(DBUS_LIBS)								\
-	$(PANGO_LIBS)								\
 	$(GMODULE_LIBS)								\
 	$(GTHREAD_LIBS)								\
 	$(GIO_LIBS)								\
diff --git a/tests/tracker-store/Makefile.am b/tests/tracker-store/Makefile.am
index 9261e17..20db3a9 100644
--- a/tests/tracker-store/Makefile.am
+++ b/tests/tracker-store/Makefile.am
@@ -19,7 +19,6 @@ INCLUDES = 									\
 	$(GCOV_CFLAGS)								\
 	$(GMODULE_CFLAGS)							\
 	$(GTHREAD_CFLAGS)							\
-	$(PANGO_CFLAGS)								\
 	$(DBUS_CFLAGS)
 
 # tracker_events_SOURCES = 							\
-- 
1.7.0.4

